src/main/scala/mesosphere/marathon/api/v2/PodsResource.scala

```diff
@@ -280,7 +280,7 @@ class PodsResource @Inject() (
       validateOrThrow(instanceId)
       val parsedInstanceId = Instance.Id.fromIdString(instanceId)
       val instances = await(taskKiller.kill(id, _.filter(_.instanceId == parsedInstanceId), wipe))
-      instances.headOption.fold(unknownTask(instanceId))(instance => ok(jsonString(instance)))
+      instances.headOption.fold(unknownTask(instanceId))(instance => ok(jsonString(state.Instance.fromCoreInstance(instance))))
```

Comment:
Following up on the discussion on core vs state instance, I guess I mainly don't like that API layer (so the highest one) must convert between those two. Ideally for me there will be an interface (e.g. repository or service or whatever) that will accept core instances and give you core instances when you ask for them and the whole fact that the serialization format is different will be hidden behind that interface. What do you think about that?

Reply:
oh ... why do we need state.Instance here anyway? Why can't we use the core instance for serialization?

Reply:
As I mentioned [above](https://github.com/mesosphere/marathon/pull/6661#discussion_r236960675). Serialization is not the issue. Deserialization is. How can the `InstanceRepository` set the run spec? It does not "know" about the `GroupRepository`.

Reply:
Just saw that this is the API layer. We could serialize core directly. Well, It should be a RAML first anyways and then serialized.

Reply:
Will be addressed in https://github.com/mesosphere/marathon/pull/6703.